### PR TITLE
Generation: past kv can be None

### LINKED ISSUE #31017: model.generate() able to accept past_key_values=None
### Feature request

In `model.generate()`, if we pass in `past_key_values=None`, it should behave as if the keyword argument is not there. 

### Motivation

Currently, if we pass in `past_key_values=None`, we would get an error in utils.py. For my use case, I am evaluating a model with prompts of varying context length, but the prefix is the same, so the code looks like this
```
   prompts = ['a', 'ab', 'abc', 'abcd', ...]
   kv_cache = None
    for i in range(len(prompts)):
        prompt = prompts[i]
        inputs = tokenizer(prompt, return_tensors="pt", padding='longest').to(config.device)
        with torch.no_grad():
            if i == 0:
                outputs = model.generate(**inputs, generation_config=generation_config)
            else:
                outputs = model.generate(**inputs, generation_config=generation_config, past_key_values=kv_cache)
```
I want to take advantage of the past_key_values, but in the first iteration of the loop, there is no `past_key_values` yet, so `kv_cache` is `None` and I have to use an if statement to check it. I wanted to avoid the if statement around `model.generate()` by directly passing `past_kv_values=kv_cache` even when the variable is None. 

### Your contribution

in `tranformers/generation/utils.py` L1317, check if `past_key_values is None`.

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Fixes #31017. The linked code would work in the prev version, so we have to check if "past_key_values" is not None when obtaining cache position.

### CODE REVIEW

**Review by gante — APPROVED** [review]
LGTM 👍

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!
